source/FreeRTOS_DNS_Parser.c

```diff
@@ -1197,7 +1202,7 @@
                             vSetField32( pxAnswer, NBNSAnswer_t, ulTTL, dnsNBNS_TTL_VALUE );
                             vSetField16( pxAnswer, NBNSAnswer_t, usDataLength, 6 );           /* 6 bytes including the length field */
                             vSetField16( pxAnswer, NBNSAnswer_t, usNbFlags, dnsNBNS_NAME_FLAGS );
-                            vSetField32( pxAnswer, NBNSAnswer_t, ulIPAddress, FreeRTOS_ntohl( *ipLOCAL_IP_ADDRESS_POINTER ) );
+                            vSetField32( pxAnswer, NBNSAnswer_t, ulIPAddress, ulLocalIPAddress );
                         #else
```

Comment:
Should we add a check for NULL prvFindEndPointOnNetMask  condition.

Reply:
> I tried this before. But I got a problem if prvFindEndPointOnNetMask returns NULL. Should we answer this query if we can't find an endpoint for it?

Hm, we can remove those tests because every incoming packet must have an endpoint assigned.
I think of this:
~~~diff
-    if( pxNetworkBuffer->pxEndPoint == NULL )
-    {
-        pxNetworkBuffer->pxEndPoint = prvFindEndPointOnNetMask( pxNetworkBuffer );
-    }
-    
-    if( pxNetworkBuffer->pxEndPoint != NULL )
-    {
-        ( void ) memcpy( &xEndPoint, pxNetworkBuffer->pxEndPoint, sizeof( xEndPoint ) );
-    }
+    configASSERT( pxNetworkBuffer->pxEndPoint != NULL );
+    ( void ) memcpy( &xEndPoint, pxNetworkBuffer->pxEndPoint, sizeof( xEndPoint ) );
~~~
and the `configASSERT()` is only usefull during the static tests.

Reply:
I commited a change to check it after prvFindEndPointOnNetMask. Please help to review again.